### PR TITLE
feat: update to use proxy for staking apr

### PR DESCRIPTION
**Pull request by rosepuppy**
Screenshot 2024-06-20 at 2 15 33 PM

---

## Views

* `src/pages/token/rewards/StakingPanel.tsx` and `src/views/dialogs/StakeDialog.tsx`
  * update hardcoded APRs

## Hooks

* `src/hooks/useStakingAPR.ts`
  * useQuery to fetch the APR proxy endpoint for rewardRate

* `src/hooks/useEndpointsConfig.ts`
  * add new `stakingAPR` endpoint to config

### CODE REVIEW

File: src/hooks/useStakingAPR.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        'content-type': 'application/json',
+      },
+    });
+
+    const data = await response.json();
+    return data?.[0].rewardRate as number | undefined;
+  };
+
+  const { data } = useQuery({
+    queryKey: ['stakingAPY'],
```
Comment (moo-onthelawn) [inline src/hooks/useStakingAPR.ts:23]: nit: 'stakingApr'

File: src/hooks/useStakingAPR.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    queryFn,
+    enabled: true,
+    refetchOnWindowFocus: false,
+    refetchOnReconnect: false,
+  });
+
+  const formatedAPR = data ? (data * 100).toFixed(2) : undefined;
+
+  return formatedAPR;
```
Comment (moo-onthelawn) [inline src/hooks/useStakingAPR.ts:32]: nit: formattedAPR

File: src/hooks/useStakingAPR.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  const { data } = useQuery({
+    queryKey: ['stakingAPY'],
+    queryFn,
+    enabled: true,
+    refetchOnWindowFocus: false,
+    refetchOnReconnect: false,
+  });
+
+  const formatedAPR = data ? (data * 100).toFixed(2) : undefined;
```
Comment (moo-onthelawn) [inline src/hooks/useStakingAPR.ts:30]: nit: can we use `PERCENT_DECIMALS` here? I wonder if it's also technically more correct to be using an `Output` component in the receivers (we could technically make an `AprTag` component that abstracts this away though I guess that adds a bit of work)

**Review by moo-onthelawn — APPROVED** [review]
woooHHoooo

File: src/hooks/useStakingAPR.ts (hunk 2 above)
Comment (rosepuppy) [reply src/hooks/useStakingAPR.ts:32]: can't spell 😅

File: src/hooks/useStakingAPR.ts (hunk 3 above)
Comment (rosepuppy) [reply src/hooks/useStakingAPR.ts:30]: I tried using output, but the spacing was a bit wierd in the tag, and had an unwanted % suffix, also not sure output is meant to be used in tags 🤔

**Pull request merged**